connect-button/src/main/java/com/ifttt/connect/ui/ConnectButton.java

```diff
@@ -120,6 +122,9 @@ public void setup(Configuration configuration) {
                 if (configuration.inviteCode != null) {
                     clientBuilder.setInviteCode(configuration.inviteCode);
                 }
+                if (configuration.credentialsProvider != null) {
+                    clientBuilder.setCredentialsProvider(configuration.credentialsProvider);
+                }
```

Comment:
This is in the if loop - so for the case where an api client wasn't passed as part of the configuration. 

For the case where the `Configuration` contains the `ConnectionApiClient` i.e. the developer passed along the client and also the `CredentialsProvider`, shouldn't we set the credentials provider to the client?

Reply:
or do we expect the api client to be built correctly i.e. by setting the Credential provider.

Reply:
We should make CredentialProvider, or UserTokenProvider, a required parameter.